public/js/controllers/dashboard.js

```diff
@@ -0,0 +1,112 @@
+angular.module('mean.system')
+  .controller('DashboardController', ['$scope', '$http', 'Global', ($scope, $http, Global) => {
+    $scope.donationData = {};
+    $scope.gamesData = {};
+    $scope.leaderBoardData = {};
+    $scope.leaderBoard = true;
+    $scope.gameLog = false;
+    $scope.donations = false;
+    $scope.noGameLog = false;
+    $scope.noDonations = false;
+    $scope.noLeaderBoard = false;
+    $scope.gameLogRange = [];
+    $scope.donationsCount = 0;
+    $scope.leaderBoardCount = 0;
+    angular.element('#leaderboard').addClass('active-class');
+
+    $scope.getDonations = (token) => {
+      token = localStorage.getItem('token');
+      $http.get(`/api/v1/donations/${token}`).then((user) => {
+        $scope.donationData = user.data;
+        const range = [];
+        for (let index = 1; index < (user.data).length + 1; index += 1) {
+          range.push(index);
+        }
+        $scope.donationsCount = range;
+      });
+    };
+
+    $scope.getGameLog = (token) => {
+      token = localStorage.getItem('token');
+      $http.get(`/api/v1/games/history/${token}`).then((user) => {
+        $scope.gamesData = user.data;
+        const range = [];
+        for (let index = 1; index < (user.data).length + 1; index += 1) {
```

Comment:
It's advisable not to use iterators (loops), but instead, Prefer JavaScript’s higher-order functions.